lfsapi/endpoint.go

```diff
@@ -106,3 +107,10 @@ func endpointFromGitUrl(u *url.URL, e *endpointGitFinder) Endpoint {
 	u.Scheme = e.gitProtocol
 	return Endpoint{Url: u.String()}
 }
+
+func endpointFromLocalPath(path string) Endpoint {
+	if !strings.HasSuffix(path, ".git") {
+		path = slashpath.Join(path, ".git")
```

Comment:
Use of `path.Join` is not recommended within Git LFS. I would recommend instead:

```go
if !strings.HasSuffix(path, ".git") {
        path = fmt.Sprintf("%s/.git", path)
}

// ...
```